weldx/asdf/file.py

```diff
@@ -0,0 +1,471 @@
+        del self._asdf_handle.tree[item]
+        self.add_history_entry(f"deleted key {item}")
+        self._asdf_handle.update()
+
+    @property
+    # TODO: should we actually expose this? It allows advanced operations for adults.
+    # TODO: return type is also not guaranteed to be IOBase, right?
+    def file_handle(self) -> IOBase:
+        if self._asdf_handle._closed:
+            raise RuntimeError("closed file, cannot access file handle.")
+        return self._asdf_handle._fd._fd
+
+    def as_attr(self) -> MutableMapping:
+        """Return the Weldx dictionary as an attributed object.
+
+        Returns
+        -------
+        attrdict :
+            This dictionary wrapped such, that all of its keys can be accessed as
+            properties.
+
+        Examples
+        --------
+        >>> tree = dict(wx_meta={"welder": "Nikolai Nikolajewitsch Benardos"})
+        >>> wf = WeldxFile(tree=tree)
+        >>> wfa = wf.as_attr()
+        >>> wfa.wx_meta.welder
+        'Nikolai Nikolajewitsch Benardos'
+
+        """
+        import attrdict
+
+        return attrdict.AttrDict(self)
+
+    def write_to(self, fd: Optional[types_path_and_file_like] = None, **asdf_args):
+        """write current weldx file to given file name.
+
+        Parameters
+        ----------
+        fd : str, pathlib.Path or file-like object
+            May be a string path to a file, or a Python file-like
+            object.  If a string path, the file is automatically
+            closed after writing.
+
+        asdf_args :
+            Allowed parameters:
+
+            * all_array_storage=None
+            * all_array_compression='input'
+            * auto_inline=None
+            * pad_blocks=False
+            * include_block_index=True
+            * version=None
+
+        Returns
+        -------
+        The given input file name or a buffer, in case the input was omitted.
+        """
+        created = False
+        if fd is None:
+            fd = BytesIO()
+            created = True
+
+        # TODO: if no args are given, should we use the args given in ctor?
+        self._asdf_handle.write_to(fd, **asdf_args)
+
+        if created:
+            fd.seek(0)
+        return fd
+
+    def view_tree(self, path: Tuple = None):
+        """Display YAML header
+
+        Parameters
+        ----------
+        path :
+            tuple representing the lookup path in the YAML/ASDF tree.
+
+        Returns
+        -------
+
+        """
+        from weldx.asdf.util import view_tree
+
+        return view_tree(self.file_handle, path=path)
+
+    def _header_as_tree(self):
+        from ipytree import Tree, Node
+
+        header = get_yaml_header(self.file_handle, parse=True)
+        tree = Tree()
+        for x in header:
+            node = Node(x)
+            tree.add_node(node)
+            for child in x:
+                child_node = Node(child)
+                node.add_node(child_node)
+        return tree
```

Comment:
I think using `ipytree` is the best way to go for more complex and interactive file browsing. I looked into it a little at the beginnig but the `JSON` display was just convenient

one idea could be to implement `_as_ipytree` methods for our custom classes like LCS, CSM etc. that attach their own subtrees with custom information

Reply:
Building the tree incrementally for different high-level objects seems like a very good idea. I was just looking for a more "interactive" widget to handle large trees. I thought we already had a widget doing this job, but was unable to find it.